### PR TITLE
Subscription status and unregister apis

### PR DESCRIPTION
**Pull request by k-antipochkin**
### Description of Changes

1. Exposed an API to return the push subscription status
2. Exposed an API to unregister from push
3. Refactored the code to use a push manager which is initialized at the very beginning

### Breaking Changes

-   None

### Release Checklist

Prepare:

-   [ ] Detail any breaking changes. Breaking changes require a new major version number

Bump versions in:

-   [ ] package.json
-   [ ] package-lock.json (run `npm install` after editing package.json)
-   [ ] src/core/index.ts

Build:

-   [ ] `npm run dist`
-   [ ] Commit resulting `dist/` changes

Release:

-   [ ] Squash and merge to master
-   [ ] Delete branch once merged
-   [ ] Create tag from master matching chosen version
-   [ ] Fill out release notes
-   [ ] Run `npm publish --access public`
-   [ ] Publish `dist/worker.js` and `dist/web-bundle.js` to CDN (re-run `npm run dist` locally to update `web-bundle.js`)

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -214,9 +219,14 @@ export default class Kumulos {
         return trackEvent(this.context, type, properties).then(_ => void 0);
     }
 
+    async getPushSubscriptionStatus(subscriptionStateListener: (subscriptionState: PushSubscriptionState) => void) {
```
Comment (cgwyllie) [inline src/index.ts:223]: Any particular reason to use a callback here instead of just returning a Promise?

File: src/prompts/index.tsx [hunk 2]
```diff
@@ -38,12 +38,13 @@ export class PromptManager {
...
     private readonly pushContainer: RootFrameContainer;
     private readonly triggerFilter: PromptTriggerEventFilter<PushPromptConfig>;
+    private readonly pushManager: PushOpsManager;
 
     private state?: PromptManagerState;
     private subscriptionState?: PushSubscriptionState;
     private activePrompts: PushPromptConfig[];
     private currentlyRequestingPrompt?: PushPromptConfig;
-    private pushOpsManager?: PushOpsManager;
+    // private pushOpsManager?: PushOpsManager;
```
Comment (cgwyllie) [inline src/prompts/index.tsx]: To be removed?

Comment (k-antipochkin) [reply src/prompts/index.tsx]: Removed

File: src/index.ts (hunk 1 above)
Comment (k-antipochkin) [reply src/index.ts:223]: The way the methods are currently cached and called requires it to be a listener. The Optimove part exposes it as  as a promise though

Comment (cgwyllie) [reply src/index.ts:223]: Ah gotcha because we use the global function invocations & not the class instance directly.

**Review by cgwyllie — DISMISSED** [review]

**Pull request closed without merging**